test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,341 @@
+	for _, pod := range pods.Items {
+		// TODO: make waiting parallel.
+		err := waitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		if err != nil {
+			e = append(e, err)
+		}
+	}
+	return e
+}
+
+func verifyPodsResponding(c *client.Client, ns, name string, pods *api.PodList) error {
+	By("trying to dial each unique pod")
+	retryTimeout := 2 * time.Minute
+	retryInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	return wait.Poll(retryInterval, retryTimeout, podResponseChecker{c, ns, label, name, pods}.checkAllResponses)
+}
+
+var _ = Describe("ResizeNodes", func() {
+	supportedProviders := []string{"gce"}
+	var testName string
+	var c *client.Client
+	var ns string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err := createTestingNS("resize-nodes", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !providerIs(supportedProviders...) {
+			return
+		}
+		By(fmt.Sprintf("destroying namespace for this suite %s", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete namespace %v", err)
+		}
+		By("restoring the original node instance group size")
+		if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %v", err)
+		}
+		if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %v", err)
+		}
+		if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original cluster size: %v", err)
+		}
+	})
+
+	testName = "should be able to delete nodes."
+	It(testName, func() {
+		Logf("starting test %s", testName)
+
+		if !providerIs(supportedProviders...) {
+			By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
+				testName, supportedProviders, testContext.Provider))
+			return
+		}
+
+		if testContext.CloudConfig.NumNodes < 2 {
+			By(fmt.Sprintf("skipping %s test, which requires at lease 2 nodes (not %d)",
+				testName, testContext.CloudConfig.NumNodes))
+			return
+		}
+
+		// Create a replication controller for a service that serves its hostname.
+		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+		name := "my-hostname-delete-node-" + string(util.NewUUID())
+		replicas := testContext.CloudConfig.NumNodes
+		createServeHostnameReplicationController(c, ns, name, replicas)
+		pods, err := waitForPodsCreated(c, ns, name, replicas)
+		Expect(err).NotTo(HaveOccurred())
+		e := waitForPodsRunning(c, pods)
+		if len(e) > 0 {
+			Failf("Failed to wait for pods running: %v", e)
+		}
+		err = verifyPodsResponding(c, ns, name, pods)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("decreasing cluster size to %d", replicas-1))
```

Comment:
You could put all of these calls within the `Expect(...)`, so that they would look like:

```go
Expect(resizeNodeInstanceGroup(replicas - 1)).NotTo(HaveOccurred())
```

However, that then reads the exact opposite of what we want, which is ultra-confusing. @roberthbailey any Golang advice here?